### ISSUE TITLE
Nightly Build Release checklist 

### ISSUE THREAD

**Issue by ofTheo** [openframeworks/openFrameworks#8192]
Following up @dimitre 

- [x] #8213
- [x] #8211
- [ ] #8124
- [ ] https://github.com/openframeworks/projectGenerator/issues/580
- [x] #8215
- [x] #8193
- [x] #8170 
- [x] #8190
- [x] #8155 
- [x] #8102 
- [x] #8194 
- [x] Need to build OpenCV with `"WITH_CAROTENE": "OFF".` 
- [x] #8273 

Please feel free to edit this issue to add  other issues that point to broken nightly builds.

**Comment by dimitre** [comment]
it seems RPI libs download is not working : 
bzip2: (stdin) is not a bzip2 file.
https://github.com/openframeworks/openFrameworks/actions/runs/11860266785/job/33055126869

**Comment by ofTheo** [comment]
@danoli3 @roymacdonald @dimitre @artificiel @oxillo 

Let's try and get these nightly builds actually working! :-) 

Would be great if you have a moment to download the nightly builds from: 
https://github.com/openframeworks/openFrameworks/releases/tag/nightly 

And try both allAddonsExample and use the PG to generate a project and see if things build successfully. 

Right now macOS is broken ( see #8170 ) but I could imagine VS / msys and others might have issues too.

**Comment by dimitre** [comment]
Actions are not working as they should. 
Libraries are pointing to 404 in half of the platforms.

**Comment by artificiel** [comment]
`of_v20241109_ios_release`: tested with macOS 14.5 on M1 with a 16.0.1 device. successfully compiled-run 3 examples (fontShapesExample, soundPlayerExample, vboExample). (not sure about allAddonsExample on this...?)

A handful of compiler warnings; nothing outrageous.
Simulators don't work, but "Designed for iPad" does.

PG 0.83.0 pointed to the correct OF tree but came up configured with the "OSX" platform, which threw me off a moment until i noticed i was compiling a mac project. once re-seated to iOS (Xcode) things worked, with a simple OSC test to the broadcast LAN.

**Comment by ofTheo** [comment]
@artificiel - that's some welcome news! I know the iOS stuff was a big focus for all the apothecary changes. 
Great to hear its all working.

**Comment by dimitre** [comment]
now pg is adding correctly xcframeworks. allAddonsExample still fails on opencv / carotene
it works if you substitute the .a file with the one from "ALTpothecary"
https://github.com/dimitre/ofChalet/releases

**Comment by dimitre** [comment]
Investigating more of the scripts to build of nightly, it seems there are a lot of scripts being called with "-b" parameter which is to enable variable BLEEDING_EDGE in ```scripts/dev/download_libs.sh```
which in turn decides if $VER variable is "latest" or "master"

Looking at apothecary libs, we have the relation between the releases and $ver variable
latest = latest
bleeding = bleeding
nightly = master

that is strange. 
but to make it work as it is we should set ```VER = $REPO``` and if repo == nightly set it to master.

**Comment by artificiel** [comment]
@dimitre haha well indeed this should be disentangled.

if we are to advertise git repo for common consumption (in coherence to "use nightlies" as discussed in another thread) it is important that it's easy to understand, as working in git means using `download_libs` periodically.
|*current name*| *use*| *download_libs*| *suggested branch rename*|*CI*|*note*
-|-|-|-|-|-
master|"stable git"| default  | stable | nightlies | default branch
bleeding|active devs|  -flag required | develop (or active) | ¿ just tests ? |
latest|latest bin|n/a|n/a| point releases

could the download_libs script be adjusted so it does the right thing by default? It can detect which branch the user is switched to. in any case the switch would follow the branch name so it's 1:1 

(I'm not sure where is the best place to have this conversation it seems like an all-encompassing topic)

**Comment by dimitre** [comment]
I think everything is more complex than it should be so in my opinion we can use only repo and set $VER accordingly.
with this plus some changes in OpenCV formula we can get nightly building to test. we can copy settings from "Altpothecary" to make it work. 
Questions: is VideoIO needed in opencvPeopleDetector?
should we remove "videoio" include and settings from apothecary?
should we guard with #if defined(HAVE_OPENCV_VIDEOIO) ?

**Comment by ofTheo** [comment]
@danoli3 - would love some input to the above - especially @artificiel 's grid. 
Would be good to know what the plan was and what we should be pointing people to.

**Comment by ofTheo** [comment]
@dimitre 

> Questions: is VideoIO needed in opencvPeopleDetector?
> should we remove "videoio" include and settings from apothecary?

I don't believe videoIO is needed for that addon. I am guessing there are a fair number of things we can turn off that might help compilation.

**Comment by roymacdonald** [comment]
As for the macos opencv issue, the problem is twofold. First, the library is compiled somehow that it is missing stuff. The [one in here](https://github.com/openframeworks/apothecary/releases/tag/bleeding) works while the one in `latest` does not.

Second, it is the project generator.
[This commit broke it](https://github.com/openframeworks/projectGenerator/commit/c6503b0cf1e46d038abd109577728a2870a68340)

That is the problem but I am not fixing it.

I am done with all this crap, I spent 2 weeks working full time in order to get PG working again and then in an attempt to fix something that was not broken it actually gets broken and apparently not tested properly. I am tired of this way of developing and I am out. 
I've been contributing to this community for at least 14 years but now it is time to move on, so this is actually a really tough decision to make. 
I will still keep on using OF but no longer count with me for its core development.

I wish you the best and I hope that OF manages to keep afloat.

**Comment by danoli3** [comment]
Settle down there Roy hahaha 😂

**Comment by ofTheo** [comment]
Getting pretty close here, the last two issues in this list are the PG issues regarding adding subfolders of src/ and local addons. 

I think in addition a couple of the actions/runners are failing. 
I can look into those.

**Comment by dimitre** [comment]
hey @ofTheo can you explain better the PG issues? which platform?
I've just tested with subfolders of src and additional sources and it is working OK here

<img width="272" alt="Screenshot 2024-11-26 at 17 29 35" src="https://github.com/user-attachments/assets/fe51b4d3-d519-45bb-b3d1-9496e04fd48b">

**Comment by dimitre** [comment]
Now I've read about this in VS. I'll take a look soon. 
I've updated with some other issues happening on nightly/latest, if others can jump and help

**Comment by dimitre** [comment]
Maybe good for closing?
